Remove support for "profile" in vault util module

rallytime opened the pull request:
This was marked for removal in the Fluorine release.

Fixes #46721

Commit:
Remove support for "profile" in vault util module

This was marked for removal in the Fluorine release.

cachedout commented on doc/topics/releases/fluorine.rst:
```diff
@@ -140,3 +140,11 @@ The ``trafficserver`` state had the following changes:
+
+The ``vault`` utils module had the following changes:
+
+- Support for specifying Vault connection data within a 'profile has been removed.
```
Missing closing quote around `profile`?

cachedout commented on doc/topics/releases/fluorine.rst:
```diff
@@ -140,3 +140,11 @@ The ``trafficserver`` state had the following changes:
+The ``vault`` utils module had the following changes:
+
+- Support for specifying Vault connection data within a 'profile has been removed.
+  Please see the vault documentation for details on the new configuration schema.
```
I'm not sure which documentation this refers to. Could we link to it?

Commit:
Remove unused import from vault utils module

Commit:
Update Fluorine release notes with link to vault module docs

Commit:
Update Fluorine release notes to use double back-ticks instead of single back-ticks

RST formatting needs double backticks to properly display `code` font markings. The
use of single backticks is used in markdown.

rallytime commented:
re-run all
